mungegithub/mungers/close-stale-pr.go

```diff
@@ -0,0 +1,302 @@
+	lastHuman := obj.Issue.CreatedAt
+
+	comments, err := obj.ListComments()
+	if err != nil {
+		return nil, err
+	}
+
+	for i := range comments {
+		comment := comments[i]
+		if !validComment(comment) {
+			continue
+		}
+		if mergeBotComment(comment) || jenkinsBotComment(comment) {
+			continue
+		}
+		if lastHuman.Before(*comment.UpdatedAt) {
+			lastHuman = comment.UpdatedAt
+		}
+	}
+
+	return lastHuman, nil
+}
+
+func findLastInterestingEventUpdate(obj *github.MungeObject) (*time.Time, error) {
+	lastInteresting := obj.Issue.CreatedAt
+
+	events, err := obj.GetEvents()
+	if err != nil {
+		return nil, err
+	}
+
+	for i := range events {
+		event := events[i]
+		if event.Event == nil || *event.Event != "reopened" {
+			continue
+		}
+
+		if lastInteresting.Before(*event.CreatedAt) {
+			lastInteresting = event.CreatedAt
+		}
+	}
+
+	return lastInteresting, nil
+}
+
+func findLastModificationTime(obj *github.MungeObject) (*time.Time, error) {
+	lastHumanIssue, err := findLastHumanIssueUpdate(obj)
+	if err != nil {
+		return nil, err
+	}
+	lastHumanPR, err := findLastHumanPullRequestUpdate(obj)
+	if err != nil {
+		return nil, err
+	}
+	lastInterestingEvent, err := findLastInterestingEventUpdate(obj)
+	if err != nil {
+		return nil, err
+	}
+
+	lastModif := lastHumanPR
+	if lastHumanIssue.After(*lastModif) {
+		lastModif = lastHumanIssue
+	}
+	if lastInterestingEvent.After(*lastModif) {
+		lastModif = lastInterestingEvent
+	}
+
+	return lastModif, nil
+}
+
+func findLatestWarningComment(obj *github.MungeObject) *githubapi.IssueComment {
+	var lastFoundComment *githubapi.IssueComment
+
+	comments, err := obj.ListComments()
+	if err != nil {
+		return nil
+	}
+
+	for i := range comments {
+		comment := comments[i]
+		if !validComment(comment) {
+			continue
+		}
+		if !mergeBotComment(comment) {
+			continue
+		}
+
+		if !warningCommentRE.MatchString(*comment.Body) {
+			continue
+		}
+
+		if lastFoundComment == nil || lastFoundComment.CreatedAt.Before(*comment.UpdatedAt) {
+			if lastFoundComment != nil {
+				obj.DeleteComment(lastFoundComment)
+			}
+			lastFoundComment = &comment
+		}
+	}
+
+	return lastFoundComment
+}
+
+func durationToDays(duration time.Duration) string {
+	days := duration / day
+	dayString := "days"
+	if days == 1 || days == -1 {
+		dayString = "day"
+	}
+	return fmt.Sprintf("%d %s", days, dayString)
+}
+
+func closePullRequest(obj *github.MungeObject, inactiveFor time.Duration) {
+	comment := findLatestWarningComment(obj)
+	if comment != nil {
+		obj.DeleteComment(comment)
+	}
+
+	obj.WriteComment(fmt.Sprintf(closingComment, durationToDays(inactiveFor)))
+	obj.ClosePR()
+}
+
+func postWarningComment(obj *github.MungeObject, inactiveFor time.Duration, closeIn time.Duration) {
+	obj.WriteComment(fmt.Sprintf(
+		warningComment,
+		durationToDays(inactiveFor),
+		durationToDays(closeIn)))
+}
+
+func checkAndWarn(obj *github.MungeObject, inactiveFor time.Duration, closeIn time.Duration) {
```

Comment:
this 'might should be' unit tested, seems like a lot going on...

admittedly sometimes we don't test at all, and sometimes we just test `Munge`..